src/compiler/emitter.ts

```diff
@@ -4502,7 +4516,195 @@ module ts {
+                    if (member.kind === SyntaxKind.MethodDeclaration || node.kind === SyntaxKind.MethodSignature) {
+                        if (!(<MethodDeclaration>member).body) {
+                            return emitPinnedOrTripleSlashComments(member);
+                        }
+
+                        writeLine();
+                        emitLeadingComments(member);
+                        emitStart(member);
+                        if (member.flags & NodeFlags.Static) {
+                            write("static ");
+                        }
+                        emitMemberAccessForPropertyName((<MethodDeclaration>member).name);
+                        emitSignatureAndBody(<MethodDeclaration>member);
+                        emitEnd(member);
+                        emitTrailingComments(member);
+                    }
+                    else if (member.kind === SyntaxKind.GetAccessor || member.kind === SyntaxKind.SetAccessor) {
+                        var accessors = getAllAccessorDeclarations(node.members, <AccessorDeclaration>member);
+                        if (member === accessors.firstAccessor) {
+                            writeLine();
+                            if (accessors.getAccessor) {
+                                emitLeadingComments(accessors.getAccessor);
+                                emitStart(accessors.getAccessor);
+                                if (member.flags & NodeFlags.Static) {
+                                    write("static ");
+                                }
+                                write("get ");
+                                emitMemberAccessForPropertyName((<MethodDeclaration>member).name);
+                                emitSignatureAndBody(accessors.getAccessor);
+                                emitEnd(accessors.getAccessor);
+                                emitTrailingComments(accessors.getAccessor);
+                            }
+                            if (accessors.setAccessor) {
+                                // We will only write new line if we just emit getAccessor
+                                emitLeadingComments(accessors.setAccessor);
+                                emitStart(accessors.setAccessor);
+                                if (member.flags & NodeFlags.Static) {
+                                    write("static ");
+                                }
+                                write("set ");
+                                emitMemberAccessForPropertyName((<MethodDeclaration>member).name);
+                                emitSignatureAndBody(accessors.setAccessor);
+                                emitEnd(accessors.setAccessor);
+                                emitTrailingComments(accessors.setAccessor);;
+                            }
+                        }
+                    }
+                });
+            }
+
+            function emitConstructorOfClass(node: ClassDeclaration, baseTypeNode: TypeReferenceNode) {

```

Comment:
i do nto think there is another kind of constructors. so would name this either "emitClassConstructor" or just "emitConstructor"